src/Diagnostics/CodeAnalysis/Core/MetaAnalyzers/RegisterActionAnalyzer.cs

```diff
@@ -7,12 +7,14 @@
 using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using System.Threading;
 
 namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers
 {
-    public abstract class RegisterActionAnalyzer<TClassDeclarationSyntax, TInvocationExpressionSyntax, TLanguageKindEnum> : DiagnosticAnalyzerCorrectnessAnalyzer
+    public abstract class RegisterActionAnalyzer<TClassDeclarationSyntax, TInvocationExpressionSyntax, TArgumentSyntax, TLanguageKindEnum> : DiagnosticAnalyzerCorrectnessAnalyzer
```

Comment:
I think I understand the rationale for adding this analysis to an existing analyzer, especially one with such an alluring name, but this analysis is distinct enough that it might have warranted a separate analyzer. I am not asking for any code change at this late hour, but it might be worth considering in a less pressured moment.